bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/ChatGPTHandler.java

```diff
@@ -175,19 +224,34 @@ public void initialize() {
                 ContentResponse response = request.send();
                 if (response.getStatus() == 200) {
                     updateStatus(ThingStatus.ONLINE);
-                    JsonObject jsonObject = gson.fromJson(response.getContentAsString(), JsonObject.class);
-                    if (jsonObject != null) {
-                        JsonArray data = jsonObject.getAsJsonArray("data");
-
-                        List<String> modelIds = new ArrayList<>();
-                        for (JsonElement element : data) {
-                            JsonObject model = element.getAsJsonObject();
-                            String id = model.get("id").getAsString();
-                            modelIds.add(id);
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    try {
+                        JsonNode models = objectMapper.readTree(response.getContentAsString());
+                        JsonNode data = models.get("data");
+
+                        if (data != null) {
+                            logger.debug("Models: {}", data.toString());
+
+                            List<String> modelList = new ArrayList<>();
+
+                            data.forEach(model -> {
+                                JsonNode id = model.get("id");
+
+                                if (id != null) {
+                                    modelList.add(id.asText());
+                                }
+                            });
+
+                            this.models = List.copyOf(modelList);
+
+                        } else {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "@text/offline.communication-error");
                         }
-                        this.models = List.copyOf(modelIds);
-                    } else {
-                        logger.warn("Did not receive a valid JSON response from the models endpoint.");
+
+                    } catch (JsonProcessingException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                        logger.error("Failed to parse models: {}", e.getMessage(), e);
```

Comment:
Is this a critical situation that users need to report as a bug? The thing status "COMMUNICATION_ERROR" rather sounds like there is an issue with the network, which would not mean that there is a bug in the code.